tests/e2e/specs/customizer/header-builder/above-header/search-icon.test.js

```diff
@@ -0,0 +1,118 @@
+import { createURL } from '@wordpress/e2e-test-utils';
+import { setCustomize } from '../../../../utils/customize';
+import { setBrowserViewport } from '../../../../utils/set-browser-viewport';
+describe( 'Customizing search icon in the above header section', () => {
+	it( 'search icon size and color for desktop should apply correctly', async () => {
+		const searchIcon = {
+			'header-desktop-items': {
+				above: {
+					above_left: {
+						0: 'search',
+					},
+				},
+			},
+			'header-mobile-items': {
+				above: {
+					above_left: {
+						0: 'search',
+					},
+				},
+			},
+			'header-search-icon-space': {
+				desktop: '60',
+				'desktop-unit': 'px',
+			},
+			'header-search-icon-color': {
+				desktop: 'rgb(173, 17, 228)',
+			},
+		};
+		await setCustomize( searchIcon );
+		await page.goto( createURL( '/' ), {
+			waitUntil: 'networkidle0',
+		} );
+
+		await page.waitForSelector( '.astra-search-icon' );
+		await expect( {
+			selector: '.astra-search-icon',
+			property: 'font-size',
+		} ).cssValueToBe(
+			`${ searchIcon[ 'header-search-icon-space' ].desktop }${ searchIcon[ 'header-search-icon-space' ][ 'desktop-unit' ] }`,
+		);
+		//Search icon color for desktop
+		await expect( {
+			selector: '.ast-header-search .ast-icon',
+			property: 'color',
+		} ).cssValueToBe( `${ searchIcon[ 'header-search-icon-color' ].desktop }`,
+		);
+	} );
+	it( 'search icon color and size for tablet should apply correctly', async () => {
+		const searchIcon = {
+			'header-mobile-items': {
+				above: {
+					above_left: {
+						0: 'search',
+					},
+				},
+			},
+			'header-search-icon-color': {
+				tablet: 'rgb(7, 37, 245)',
+			},
+			'header-search-icon-space': {
+				tablet: '50',
+				'tablet-unit': 'px',
+			},
+		};
+		await setCustomize( searchIcon );
+		await page.goto( createURL( '/' ), {
+			waitUntil: 'networkidle0',
+		} );
+		await setBrowserViewport( 'medium' );
+		await expect( {
+			selector: '.ast-icon',
+			property: 'color',
+		} ).cssValueToBe( `${ searchIcon[ 'header-search-icon-color' ].tablet }`,
+		);
+		//Search icon size for tablet
+		await setBrowserViewport( 'medium' );
```

Comment:
```suggestion
		await setBrowserViewport( 'medium' );
		await expect( {
			selector: '.ast-icon',
			property: 'color',
		} ).cssValueToBe( `${ searchIcon[ 'header-search-icon-color' ].tablet }`,
		);
```
@bsfsnehal I think loading this function only once should resize the browser and we do not need to again write setBrowserViewport() function. We can just define medium, small once and then write all the cases related to that device below that.

Reply:
Yes, After removing  setBrowserViewport() function from responsive test case assertion, Still it is executing fine on local.
I update this.